src/ros_can_nodes/include/RosCanNode.hpp

```diff
@@ -11,42 +11,28 @@
 #ifndef ROSCANNODE_H
 #define ROSCANNODE_H
 
-#include "ros_node_lib/common.h"
-#include "ros_node_lib/poll_manager.h"
-#include "ros_node_lib/connection_manager.h"
-#include "ros_node_lib/topic_manager.h"
-#include "ros_node_lib/xmlrpc_manager.h"
-#include "ros_node_lib/callback_queue_interface.h"
-#include "ros_node_lib/rosout_appender.h"
-#include "ros_node_lib/publisher.h"
-#include "ros_node_lib/advertise_options.h"
-#include "ros_node_lib/callback_queue.h"
+#include "ros_node_lib/RosNode.hpp"
 #include <thread>
 #include <bitset>
 #include <unordered_map>
 #include <mutex>
 #include <utility>
-#include <ros_type_introspection/ros_introspection.hpp>
+#include <boost/shared_ptr.hpp>
 #include <topic_tools/shape_shifter.h>
 
-
-#define MAX_NODES 16
 #define MAX_TOPICS 128
 
 namespace roscan {
 
-    class NodeBackingCollection;
+    class RosCanNode;
+    typedef boost::shared_ptr<RosCanNode> RosCanNodePtr;
 
-    class RosCanNode : public boost::enable_shared_from_this<RosCanNode> {
+    class RosCanNode : public RosNode {
         public:
             RosCanNode(const std::string& name, const uint8_t id = 0);
-            ~RosCanNode();
+            ~RosCanNode() {};
```

Comment:
You should be able to remove the deceleration completely if its a default deconstructor that does nothing. 
```suggestion
```